quint/test/types/inferrer.test.ts

```diff
@@ -159,6 +159,102 @@ describe('inferTypes', () => {
     ])
   })
 
+  it('infers types for match expression', () => {
+    const defs = ['type T = A(int) | B', 'val a = variant("A", 3)', 'val nine = match a { A(n) => n * n | B => 9 }']
+
+    const [errors, types] = inferTypesForDefs(defs)
+    assert.isEmpty(errors, `Should find no errors, found: ${[...errors.values()].map(errorTreeToString)}`)
+
+    const stringTypes = Array.from(types.entries()).map(([id, type]) => [id, typeSchemeToString(type)])
+    assert.sameDeepMembers(stringTypes, [
+      [14n, 'str'],
+      [15n, 'int'],
+      [16n, '(A(int))'],
+      [17n, '(A(int))'],
+      [10n, 'str'],
+      [11n, '{}'],
+      [12n, '(B({}))'],
+      [13n, '(B({}))'],
+      [18n, '(A(int))'],
+      [24n, 'str'],
+      [26n, 'int'],
+      [19n, 'int'],
+      [20n, 'int'],
+      [21n, 'int'],
+      [25n, '(int) => int'],
+      [27n, 'str'],
+      [29n, '{}'],
+      [22n, 'int'],
```

Comment:
using concrete identifiers in this test looks a bit fragile to me. Also, what is `{}` in this context? Is it the empty record = the unit type, or is it an empty row?

Reply:
It's fragile but important, until we find a better way to do this. This is also implicitly a test for the hover functionality in LSP, ensuring with high granularity that all types are as expected. It's pretty annoying when something about ids change, but I have had errors flagged by this type of tests a lot of times already and it's often helpful (i.e. if we expected `(int) => int` and got `(_t123) => int`, probably some substitution was not properly applied).

I'd like to keep this.

Reply:
Also, for @konnov's question, types and rows are completely separate things in quint, and these values are being parsed as types, so there is no way this could be a row. I'm just trying to help disambiguate things.